### PR TITLE
Scalability Automation Review.

### PR DESCRIPTION
**Pull request by newtewt**
# Description

Hello, looking to get feedback! This is pretty much done and executing as detailed in the notes.  I still have a few things to work out. 

Created github action workflow that starts an ec2 instance, installs almost all required softwares for medic-os to run. Loads standard config and seed data through medic conf. Waits for sentinel to process. Reads the seq numbers on the views, stages an update and confirms the seq numbers bumped. 

Starts a second ec2 instance. Installs all required softwares for jmeter. Executes jmeter scalability suite. 

TODO:
Get the results into s3.
Release Medic-conf and not use a branch. 
Maybe move the csvs to the standard config folder? Not sure how we feel about that. 
Do some detailed doc on how this works. Probably a thorough google doc or maybe a bunch more additions to the Readme.md

https://github.com/medic/cht-core/issues/6266

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: .github/workflows/scalability.yml
Comment (garethbowen) [inline .github/workflows/scalability.yml:6]: Any thoughts on when we should trigger this? Is there a manual trigger option? Or maybe trigger it when we push a beta tag...

File: tests/scalability/run_suite.sh
Comment (garethbowen) [inline tests/scalability/run_suite.sh:4]: NB: branch name needs to change (obviously). I presume we can pass this as a param from GH Actions?

File: tests/scalability/start_ec2_medic.sh
Comment (garethbowen) [inline tests/scalability/start_ec2_medic.sh:114]: Automated upgrade test FTW!

File: .github/workflows/scalability.yml
Comment (newtewt) [reply .github/workflows/scalability.yml:6]: I was thinking to just trigger immediately when the beta tag gets created. There is an option to make it manual with github actions with minimal effort.

https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/

File: tests/scalability/run_suite.sh
Comment (newtewt) [reply tests/scalability/run_suite.sh:4]: Yeah, this should be the beta tag I assume. I was using my branch specifically because I had some code and csvs. Would be passed similarly as how I'm applying the branch in another area. Will update.

File: tests/scalability/start_ec2_medic.sh
Comment (dianabarsan) [inline tests/scalability/start_ec2_medic.sh:136]: NICE!!!!

File: tests/scalability/start_ec2_medic.sh
Comment (newtewt) [reply tests/scalability/start_ec2_medic.sh:82]: Yeah there are no attachments.  I'd think that'd be something we'd want to check as it affects replication time.

Comment (dianabarsan) [reply tests/scalability/start_ec2_medic.sh:82]: I wrote this script a while ago, I gave it a turn and it seems like it still works. 

https://github.com/dianabarsan/generate-form-attachments/blob/master/src/index.js
You just need to export a COUCH_URL and run it. It will create attachments for every report in the database (albeit not correctly most likely).

Comment (newtewt) [reply tests/scalability/start_ec2_medic.sh:82]: I think that would be fine. My quick read/understanding makes it seem like the attachment should be the same each time. Is that correct?

Comment (dianabarsan) [reply tests/scalability/start_ec2_medic.sh:82]: Same for the same form, yea. I basically transpiles the `fields` property into an XML and wraps it in what enketo expects.

Comment (dianabarsan) [reply tests/scalability/start_ec2_medic.sh:82]: If you're willing to use this, I can perfect it, unit test it and everything :)

Comment (newtewt) [reply tests/scalability/start_ec2_medic.sh:82]: Yeah if you want to add it to the never ending list that'd be great.

Comment (garethbowen) [reply tests/scalability/start_ec2_medic.sh:82]: @dianabarsan How did you get on with this? Should we create a new issue to add attachments after this is merged?

**Review by garethbowen — APPROVED** [review]
Looks great. Ready to merge when medic-conf is released, and we can add report attachments later.

File: tests/scalability/start_ec2_medic.sh
Comment (dianabarsan) [reply tests/scalability/start_ec2_medic.sh:82]: I wrote a script 2 weeks ago and passed it to @newtewt . 
https://github.com/medic/support-scripts/tree/master/generate-form-attachments

@newtewt did you give it a shot? Would you like me to integrate it ?

Comment (newtewt) [reply tests/scalability/start_ec2_medic.sh:82]: Yeah I had gotten time to review how it works yesterday. I'll add it today.

**Comment by newtewt** [comment]
Adds scripts to start ec2 instanes and execute our jmeter scalability tests against eachother. Also moves the seed data to the scalabaility tests directory. Adds github action workflow to execute on beta tag push.

**Pull request merged**
